### PR TITLE
Update getting started guide

### LINKED ISSUE #74: Avoid loading dynamic code out of `node_modules` directly
In the example code you showed loading the js file by getting its uri directly from the `node_modules` folder. That should be avoided, as we want extensions to be bundled and not ship their raw `node_modules` folder. So we should look into alternative ways of loading the library -- importing the js as a module or something along those lines. Though I'm not really thrilled with importing the lib directly into the extension, since it then adds size to the extension rather than just to the webview. So maybe there needs to be a bundle/webpack step or something that adds it in.

### LINKED ISSUE #348: Add docs on how to register toolkit web components in JS/TS
### Feature/component description

Based on [this issue](https://github.com/microsoft/vscode-webview-ui-toolkit/issues/347), we should do a better job of documenting/highlighting how to register toolkit web components from within a JavaScript or TypeScript file.

Like so:

```ts
import { 
  provideVSCodeDesignSystem, 
  vsCodeButton, 
  vsCodeCheckbox,
  // Import other component functions –– always in the form `vsCode{ComponentName}`
} from "@vscode/webview-ui-toolkit"

provideVSCodeDesignSystem()
  .register(
    vsCodeButton(), 
    vsCodeCheckbox(),
    // Register other component functions here
  );
```

### Use case

Toolkit users may not always want to use the toolkit via the bundled `toolkit.js` or `toolkit.min.js` files, so we should better document how to import and register individual components.

### PR DESCRIPTION
**Pull request by hawkticehurst**
### Link to relevant issue

This pull request resolves #74
This pull request resolves #348

### Description of changes

Updates the getting started guide to include new content on how to:

- Avoid loading the toolkit package directly from `node_modules`
- Register the toolkit components in JS/TS
- Add proper content security policies for loading scripts, styles, and images

### Misc

_There is also a [sibling PR](https://github.com/microsoft/vscode-webview-ui-toolkit-samples/pull/126) in the [toolkit samples repo](https://github.com/microsoft/vscode-webview-ui-toolkit-samples) that implements these changes into the toolkit extension samples._

### CODE REVIEW

**Comment by hawkticehurst** [comment]
@daviddossett these new docs could probably use another round or two of editing so whenever you have some free time I'd appreciate a pass on the content for any glaring typos, grammar, phrasing issues, etc.

File: docs/getting-started.md [hunk 1]
```diff
@@ -226,57 +226,104 @@ With an extension created, install the toolkit package using this command:
...
 
-### Using the toolkit inside a webview
+### Register the toolkit components
 
-With the package installed, you need to adjust the project so the toolkit is usable within your webview. To do this, start by updating the `_getWebviewContent` method to accept two new parameters.
+It can be helpful to think of webviews as having a frontend and a backend.
 
-```typescript
-// file: src/panels/HelloWorldPanel.ts
+In the first part of this guide, you created the backend of a webview (i.e. `HelloWorldPanel`) and in this part, you'll create and configure the frontend JavaScript of a webview.
```
Comment (daviddossett) [inline docs/getting-started.md:233]: I wonder how, if at all, we can better explain why we are framing this as a backend concept given that `HelloWorldPanel` sounds like UI.

**Review by daviddossett — APPROVED** [review]

File: docs/getting-started.md (hunk 1 above)
Comment (hawkticehurst) [reply docs/getting-started.md:233]: Agreed. This is something I'm 50-50 on including and wanted to come back to.

**Comment by OliverMKing** [comment]
I'm excited for this update but wanted to mention that I reproduced the steps and got an error on Stage 2.

`Uncaught ReferenceError: module is not defined` is shown when trying to load webview.js when running the extension. Any ideas for fixes @hawkticehurst?

**Comment by hawkticehurst** [comment]
> I'm excited for this update but wanted to mention that I reproduced the steps and got an error on Stage 2.
> 
> `Uncaught ReferenceError: module is not defined` is shown when trying to load webview.js when running the extension. Any ideas for fixes @hawkticehurst?

Hey @OliverMKing, thanks for chiming in! Alas I've run into this problem too and it's one of a handful of issues that have been blocking the release of these docs 😪

I _thinkkk_ the issue is caused by how we are bundling the toolkit and that not playing nicely once you start trying to use the components in an ESM context. Sadly, I've been pulled away to do other work in the last handful of weeks so it might take a while to circle back and get into the weeds on this one. 

For the time being I'd probably stick with the original docs –– although the writing on content security policies should still be valid/helpful!

**Comment by hawkticehurst** [comment]
@OliverMKing revisiting this PR and I believe an accidental regression caused the issue where the `"type": "module"` field was removed from `package.json`, meaning the whole toolkit was using CommonJS instead of ESM 😅

PRs #421 and #436 should have fixed these issues now, so I'd love to know if this addresses your problems.

I've also been able to continue (now successfully) [updating the sample extensions](https://github.com/microsoft/vscode-webview-ui-toolkit-samples/pull/126) to demonstrate registering the components in typescript. They're still a work in progress, but I have completed the hello world samples, so you're welcome to reference them if you'd like:

- [Hello world sample (esbuild-based)](https://github.com/microsoft/vscode-webview-ui-toolkit-samples/tree/780dd005b820c00340fe72a76a50099c5d0ef952/default/hello-world)
- [Hello world sample (webpack-based)](https://github.com/microsoft/vscode-webview-ui-toolkit-samples/tree/780dd005b820c00340fe72a76a50099c5d0ef952/default/hello-world-webpack)

File: docs/getting-started.md [hunk 2]
```diff
@@ -348,44 +567,82 @@ public static render(extensionUri: vscode.Uri) {
...
+```
+
+### Make sure everything works
+
+It's time to run a test to see if everything works. Just as before, open the extension in the Extension Development Host window by pressing `F5` and then run the `Hello World` command.
+
 ![Testing That The Toolkit Works](./assets/images/toolkit-button-test.gif)
 
-Theming is built right into the components, so you can test that out, too. Open the Command Pallette (`Crtl + Shift + P` or `Cmd + Shift + P` on Mac), search for "Preferences: Color Theme," and cycle through all the themes to see the button change.
+Theming is built right into the components, so you can test that out too. Open the Command Palette (`Crtl + Shift + P` or `Cmd + Shift + P` on Mac), search for "Preferences: Color Theme," and cycle through all the themes to see the button change.
```
Comment (daviddossett) [inline docs/getting-started.md:645]: nit: there is a chord to access this directly. Up to you if you want to reference this vs. the two step way via the command palette.
CleanShot 2023-01-17 at 15 06 21@2x

**Review by daviddossett — APPROVED** [review]

File: docs/getting-started.md [hunk 3]
```diff
@@ -10,16 +10,14 @@ _If you get stuck at any point or if you just want a pre-configured starter exte
 
 ## Part 1: Create a webview-based extension
 
-Before installing the toolkit, you need to create a webview-based extension to use the toolkit in. The following steps are taken directly from the Visual Studio Code [Your First Extension](https://code.visualstudio.com/api/get-started/your-first-extension) guide and [Webview API](https://code.visualstudio.com/api/extension-guides/webview) guide.
+Before installing the toolkit, you need to create a webview-based extension. The following steps are taken directly from the Visual Studio Code [Your First Extension](https://code.visualstudio.com/api/get-started/your-first-extension) guide and [Webview API](https://code.visualstudio.com/api/extension-guides/webview) guide.
```
[the remark this answers is not in the document]
Comment (hawkticehurst) [reply docs/getting-started.md:13]: Oh yeah, agreed. That's a good distinction to make

File: docs/getting-started.md (hunk 2 above)
Comment (hawkticehurst) [reply docs/getting-started.md:645]: Oh TIL. Definitely will add it!

Comment (hawkticehurst) [reply docs/getting-started.md:645]: Actually, scratch that, I have a GIF that demos using the two-step method and I don't really feel like re-recording/re-editing so I'll leave it for now 😅

**Pull request merged**